### PR TITLE
Deduplicate Signals: Provide Hide Functionality.

### PR DESCRIPTION
**Pull request by lennartclaas**
### Summary of changes

Enabled hide feature for merged signals in PackageCard context menu as well as in AttributionColumn.

### Context and reason for change

Is part of the story 'Deduplicate reused signals that are identical'.

### How can the changes be tested

With new tests in `audit-view.test.tsx` as well as`context-menu-audit-view.test.tsx` and with the test file `opossum-input.json` (merged signal attached to `menu.ts`).

### CODE REVIEW

File: src/Frontend/Components/AttributionColumn/AttributionColumn.tsx [hunk 1]
```diff
@@ -247,6 +250,12 @@ export function AttributionColumn(props: AttributionColumnProps): ReactElement {
     !props.displayPackageInfo.firstParty &&
     !props.displayPackageInfo.excludeFromNotice;
 
+  const attributionIdsToResolveOrUnresolve = isDisplayPackageInfo(
+    temporaryPackageInfo
+  )
+    ? temporaryPackageInfo.attributionIds
+    : [selectedPackageId];
```
Comment (benedikt-richter) [inline src/Frontend/Components/AttributionColumn/AttributionColumn.tsx:257]: Why do we need the second case (`[selectedPackageId]`)? Attributions cannot be hidden.

Comment (lennartclaas) [reply src/Frontend/Components/AttributionColumn/AttributionColumn.tsx:257]: Changed it to an empty list with the comment to consider this when refactoring. Using `const attributionIdsToResolveOrUnresolve = temporaryPackageInfo?.attributionIds ?? []` was not possible because the type checker is complaining in this case.

**Review by MarkusObendrauf — CHANGES_REQUESTED** [review]

File: src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts [hunk 2]
```diff
@@ -119,29 +119,35 @@ export function getFirstPartyChangeHandler(
...
+        for (const attributionId of attributionIds) {
+          dispatch(addResolvedExternalAttribution(attributionId));
+        }
       }
       dispatch(saveManualAndResolvedAttributionsToFile());
     }
   };
 }
 
 export function selectedPackageIsResolved(
```
Comment (benedikt-richter) [inline src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts:144]: The name seems outdated

File: src/Frontend/Components/PackageCard/PackageCard.tsx [hunk 3]
```diff
@@ -152,6 +152,10 @@ export function PackageCard(props: PackageCardProps): ReactElement | null {
       ? getPackageCardHighlighting(manualAttributions[attributionId])
       : undefined;
 
+  const displayAttributionIds = isDisplayPackageInfo(props.packageInfo)
+    ? props.packageInfo.attributionIds
+    : [attributionId];
```
Comment (benedikt-richter) [inline src/Frontend/Components/PackageCard/PackageCard.tsx:157]: Same issue as above

**Review by benedikt-richter — CHANGES_REQUESTED** [review]

File: src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts [hunk 4]
```diff
@@ -119,29 +119,35 @@ export function getFirstPartyChangeHandler(
...
-  attributionId: string,
+  attributionIds: Array<string>,
   resolvedExternalAttributions: Set<string>,
   dispatch: AppThunkDispatch
 ): () => void {
   return (): void => {
-    if (attributionId) {
-      if (resolvedExternalAttributions.has(attributionId)) {
-        dispatch(removeResolvedExternalAttribution(attributionId));
+    if (attributionIds) {
```
Comment (benedikt-richter) [inline src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts:127]: This check should be removed, as it cannot occur.

File: src/Frontend/Components/AttributionColumn/__tests__/attribution-column-helpers.test.ts [hunk 5]
```diff
@@ -29,19 +29,31 @@ describe('The AttributionColumn helpers', () => {
...
   });
 
   it('selectedPackageIsResolved returns false if empty attributionId', () => {
-    expect(selectedPackageIsResolved('', new Set<string>().add('123'))).toEqual(
-      false
-    );
+    expect(
+      selectedPackagesAreResolved([''], new Set<string>().add('123'))
+    ).toEqual(false);
+  });
```
Comment (MarkusObendrauf) [inline src/Frontend/Components/AttributionColumn/__tests__/attribution-column-helpers.test.ts:40]: do we need this test? is it expected that `attributionIds` can contain empty strings? i would expect the array to be empty instead

**Review by MarkusObendrauf — APPROVED** [review]

File: src/Frontend/Components/AttributionColumn/__tests__/attribution-column-helpers.test.ts (hunk 5 above)
Comment (lennartclaas) [reply src/Frontend/Components/AttributionColumn/__tests__/attribution-column-helpers.test.ts:40]: Only if the input file is flawed or due to some other bug. I think it should be kept. Runtime is negligible.
